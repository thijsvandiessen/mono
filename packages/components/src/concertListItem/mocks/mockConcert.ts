import type { Concert } from '@mono/data'

export const mockConcert: Concert = {
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [
    {
      startTime: '2023-09-01T13:32:36+01:00',
      id: 'test id',
      title: 'test',
      address: 'test',
      ticketLink: 'https://example.com',
      lat: 0,
      lng: 0,
    },
  ],
  url: 'some-slug',
  content: [],
}
