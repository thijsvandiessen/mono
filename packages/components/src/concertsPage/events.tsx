import React from 'react'
import classNames from 'classnames'
import styles from './styles.module.scss'
import { LoadMoreEvents } from '../loadMoreEvents/index.js'
import { Event } from '../event/index.js'
import type { EventsProps } from './concertsPage.types.js'

export const Events = ({ eventData, first, numberOfEvents }: EventsProps) => (
  <div className={classNames(styles.concerts, 'content-layout')}>
    {eventData?.map((event) => {
      if (!event?.id) return

      return (
        <Event className="content-layout--small" key={event.id} id={event.id} />
      )
    })}

    <LoadMoreEvents initialSkip={first} numberOfEvents={numberOfEvents} />
  </div>
)
