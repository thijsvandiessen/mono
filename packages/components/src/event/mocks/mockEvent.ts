import type { Event } from '@mono/graphql'

export const mockEvent: Event = {
  id: 'id',
  title: 'name',
  image: {
    id: 'image-id',
    title: 'title',
    description: 'desc',
    url: 'https://creaorkest.nl/some-image.png',
  },
  locations: [{ startTime: '2023-09-01T13:32:36+01:00', id: 'test id' }],
  url: 'some-slug',
  content: [],
}
