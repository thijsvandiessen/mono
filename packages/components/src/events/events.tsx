import React from 'react'
import { Event } from '../event/event'
import { getEvents } from '@crea/graphql/src/getters/getEvents'

export interface Props {
  skip: number
  first: number
}

export const Events = async ({ skip, first }: Props) => {
  const { data } = await getEvents({ skip, first })
  if (!data) return null

  return (
    <section>
      {data.map((event) => {
        if (!event?.id) return
        return <Event key={event.id} id={event.id} />
      })}
    </section>
  )
}
