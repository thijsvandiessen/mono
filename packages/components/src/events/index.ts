export { Events } from './events.js'
