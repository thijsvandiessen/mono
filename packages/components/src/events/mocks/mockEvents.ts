import type { Event } from '@mono/datocms'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  url: '/concerten/concert-1',
  image: undefined,
  locations: [],
}
