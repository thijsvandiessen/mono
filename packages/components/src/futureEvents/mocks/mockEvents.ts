import { type Event } from '@crea/graphql/src/types/event'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  image: undefined,
  locations: [],
  persons: [],
  program: [],
  url: '/some-slug',
  content: [],
}
