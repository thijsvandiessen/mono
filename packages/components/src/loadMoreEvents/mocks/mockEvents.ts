import { type Event } from '@mono/datocms'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  url: '/url',
  image: undefined,
  locations: [],
}
