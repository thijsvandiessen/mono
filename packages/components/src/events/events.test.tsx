import React from 'react'
import type { Props as EventProps } from '../event'
import { Events, type Props } from './events'
import { getEvents } from '@mono/datocms'
import { mockEvent } from './mocks/mockEvents'
import { resolvedComponent } from '@mono/utils'
import { render, screen } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'

vi.mock('@mono/datocms', () => ({
  ...vi.importActual('@mono/datocms'),
  __esModule: true,
  getEvents: vi.fn(),
  ConcertModelOrderBy: vi.fn(),
}))

vi.mock('../event/event', () => ({
  ...vi.importActual('../event/event'),
  __esModule: true,
  Event: (props: EventProps) => (
    <div>
      [Event]: <span>{props.id}</span>
    </div>
  ),
}))

const getEventsMock = vi.mocked(getEvents)

describe('Events component', () => {
  it('shows all the data', async () => {
    getEventsMock.mockResolvedValue({
      data: [mockEvent],
      error: undefined,
    })

    const ResolvedEvents = await resolvedComponent<Props>(Events, {
      skip: 0,
      first: 3,
    })

    const { container } = render(<ResolvedEvents />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('mock-id')).toBeInTheDocument()
  })
})
