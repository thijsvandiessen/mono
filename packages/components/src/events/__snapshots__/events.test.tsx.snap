// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Events component > shows all the data 1`] = `
<div>
  <section>
    <div>
      [Event]: 
      <span>
        mock-id
      </span>
    </div>
  </section>
</div>
`;
